Bump version to v5.3.1

andreas-bok-sociomantic opened the pull request:
Following the libtool guideline for updating
library version info, the ABI revision
number was incremented, resulting in the new
library version `9.1.0` (`libdrizzle-redux.so.9.0.1`)

For further details on libtool version-info
see: https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html

andreas-bok-sociomantic commented:
blocked on #72

andreas-bok-sociomantic commented:
@leandro-lucarella-sociomantic, agree on the bumping of the library version to 9:1:0? 

Followed the guideline here, ad 3, https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
> 3. If the library source code has changed at all since the last update, then increment revision (‘c:r:a’ becomes ‘c:r+1:a’).

leandro-lucarella-sociomantic commented:
What have changed exactly in 5.3.0?

andreas-bok-sociomantic commented:
> What have changed exactly in 5.3.0?
https://github.com/sociomantic-tsunami/libdrizzle-redux/releases/tag/v5.3.0

Changes to source code, but no additions or removals on the API. 
The behavior of the binlog has been changed which suggested a minor-bump instead of a patch-bump

leandro-lucarella-sociomantic commented:
> The behavior of the binlog has been changed which suggested a minor-bump instead of a patch-bump

Mmm, so this change is backwards incompatible? Can't you add a new symbol with the new behaviour and leave the old one with the same behaviour as before in that case?

ben-palmer-sociomantic commented:
> Mmm, so this change is backwards incompatible? Can't you add a new symbol with the new behaviour and leave the old one with the same behaviour as before in that case?

As I understood it a new feature has been added but it is backwards compatible with the old version. That to me indicates a minor bump.

andreas-bok-sociomantic commented:
> As I understood it a new feature has been added but it is backwards compatible with the old version. That to me indicates a minor bump

Yes, this is the case. A new feature which could also be considered a bug fix, to an existing feature, i.e. binlog with a non-blocking connection.

leandro-lucarella-sociomantic commented:
Then LGTM.

andreas-bok-sociomantic commented:
btw, should the project version be bumped to 5.3.1? It should have been part of the v5.3.0 release but was forgotten.

ben-palmer-sociomantic commented:
> btw, should the project version be bumped to 5.3.1? It should have been part of the v5.3.0 release but was forgotten.

Yep, so we should probably have a 5.3.1 milestone as well.

andreas-bok-sociomantic commented:
@ben-palmer-sociomantic, Ok, branch mess should be fixed. Ready for merge

andreas-bok-sociomantic commented:
Fixed errors 
```diff
diff --git a/configure.ac b/configure.ac
index 1cd21d4..4053dbb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6,7 +6,7 @@
 # Use and distribution licensed under the BSD license.  See
 # the COPYING file in this directory for full text.
 
-AC_INIT([libdrizzle-redux],[5.3.0],[https://github.com/sociomantic-tsunami/libdrizzle-redux/issues],[libdrizzle-redux],[https://github.com/sociomantic-tsunami/libdrizzle-redux])
+AC_INIT([libdrizzle-redux],[5.3.1],[https://github.com/sociomantic-tsunami/libdrizzle-redux/issues],[libdrizzle-redux],[https://github.com/sociomantic-tsunami/libdrizzle-redux])
 
 AC_CONFIG_AUX_DIR([build-aux])
 
diff --git a/rpm/spec.in b/rpm/spec.in
index ff94bda..b4abfaf 100644
--- a/rpm/spec.in
+++ b/rpm/spec.in
@@ -83,6 +83,9 @@ mkdir -p $RPM_BUILD_ROOT/
 %{_includedir}/libdrizzle-5.1/visibility.h
 
 %changelog
+* Fri September 21 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.1 
+- Bump versions for patch release v5.3.1
+
 * Fri September 16 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.0
 
 * Wed April 1 2016 Ben Palmer <ben.palmer@sociomantic.com> - 5.2.0

```

leandro-lucarella-sociomantic commented:
> btw, should the project version be bumped to 5.3.1? It should have been part of the v5.3.0 release but was forgotten.

If this is just considered a bug fix, then yes, but judging from the ABI version bump it looks like it's a new feature, so v5.4.0 then?

leandro-lucarella-sociomantic commented on rpm/spec.in:
```diff
@@ -83,5 +83,10 @@ mkdir -p $RPM_BUILD_ROOT/
 
 %changelog
+* Fri September 21 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.1 
+- Bump versions for patch release v5.3.1
```
This last line is kind of too redundant and not really saying what changed in this version :)

ben-palmer-sociomantic commented:
> If this is just considered a bug fix, then yes, but judging from the ABI version bump it looks like it's a new feature, so v5.4.0 then?

Ah its a confusing situation. It is a bug fix, but the bug fix is fixing the ABI version information.

andreas-bok-sociomantic commented:
> Ah its a confusing situation. It is a bug fix, but the bug fix is fixing the ABI version information.

> If this is just considered a bug fix, then yes, but judging from the ABI version bump it looks like it's a new feature, so v5.4.0 then?

I don't consider this as a new feature. No "source code" which alters the functionality of the library has changed.

andreas-bok-sociomantic commented:
```diff
diff --git a/rpm/spec.in b/rpm/spec.in
@@ -83,5 +83,10 @@ mkdir -p $RPM_BUILD_ROOT/
 %{_includedir}/libdrizzle-5.1/visibility.h
 
 %changelog
* Fri September 21 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.1 
-- Bump versions for patch release v5.3.1
```

leandro-lucarella-sociomantic commented:
> * Fri September 21 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.1 -- Bump versions for patch release v5.3.1

If that's supposed to be a changelog, shouldn't it say what actually changed?

andreas-bok-sociomantic commented:
> If that's supposed to be a changelog, shouldn't it say what actually changed?

Good point :) I will update both of these  

* Fri September 21 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.1 

* Fri September 16 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.0

andreas-bok-sociomantic committed:
Update changelog for v5.3.1

- Change changelog format to follow style guide at
  http://keepachangelog.com/en/0.3.0/
- Update changelog for previous releases
- Add entry for v5.3.1

andreas-bok-sociomantic committed:
Bump versions for patch release v5.3.1

Increment API version to v5.3.1

Following the libtool guideline for updating
library version info, the ABI revision
number was incremented, resulting in the new
library version `9.1.0` (so.9.0.1)

For further details on libtool version-info
see: https://www.gnu.org/software/libtool/
manual/html_node/Updating-version-info.html

andreas-bok-sociomantic commented:
Updated changelog in rpm packaging script

```diff
diff --git a/rpm/spec.in b/rpm/spec.in
index a66351a..7e11d50 100644
--- a/rpm/spec.in
+++ b/rpm/spec.in
@@ -83,9 +83,50 @@ mkdir -p $RPM_BUILD_ROOT/
 %{_includedir}/libdrizzle-5.1/visibility.h
 
 %changelog
-* Fri September 21 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.1 
-
-* Fri September 16 2016 Andreas Bok Andersen <andreas.bok@sociomantic.com> - v5.3.0
-
-* Wed April 1 2016 Ben Palmer <ben.palmer@sociomantic.com> - 5.2.0
-- Project Moving
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/)
+and this project adheres to [Semantic Versioning](http://semver.org/).
+
+## [v5.3.1] - 2016-09-21
+### Changed
+- Minor fixes to documentation, syntax and formatting
+- Fix library version for deb packaging
+- Update current and previous changelog for rpm packaging
+
+## [v5.3.0] - 2016-09-16
+### Added
+- Modify drizzle_binlog_start() so binlog data can be read with a non-blocking
+  connection
+
+### Changed
+- Update hardcoded repo location
+- Fix version number in configure script
+- Update readme and doc
+- Change location of generated documentation to top-level folder sphinx-build
+- Restructure man page doc generation
+- Bugfix where calls to drizzle_connect() would always block regardless of the
+  non_blocking option being set to true
+- Optimization of drizzle_field_buffer() so an array of drizzle_field_t pointers
+  are used to buffer fields
+- Minor fixes to .gitignore
+- Fixed integer underflow when processing large result sets using prepared
+  statements
+- Use func macro when logging functions
+- Update README with reasons for a sociomantic-tsunami fork
+- Whitespace fixes
+- Add continuous integration testing for clang compiler
+- Fix linter warnings in bootstrap.sh
+- Fix travis-ci build config for environment variables
+- Add travis-ci build status to README
+- Add unittest for event_watch_fn callback function
+- Improve documentation for binlog code example
+
+## [v5.2.0] - 2016-04-01
+### Added
+- Add callback function for poll events to client api
+
+### Changed
+- Move project repo to https://github.com/sociomantic-tsunami/libdrizzle-redux
+- Remove obsolete dependencies from deb packages
+- Rename library to libdrizzle-redux

```

ben-palmer-sociomantic commented:
LGTM

ben-palmer-sociomantic commented:
This pull request has been rebased via `git hub pull rebase`. Original pull request HEAD was 434d91a9f937ff98814324da25d2f18b5e640fcd, new (rebased) HEAD is c07059be4eb5027c593391848165141c4ef4a655
